test/cpp/api/functional.cpp

```diff
@@ -73,6 +73,19 @@ TEST_F(FunctionalTest, CosineSimilarity) {
   ASSERT_TRUE(output.allclose(expected, 1e-04));
 }
 
+TEST_F(FunctionalTest, SoftMarginLoss) {
+  auto input = torch::tensor({2., 4., 1., 3.}, torch::requires_grad());
+  auto target = torch::tensor({-1., 1., 1., -1.}, torch::kFloat);
+  auto output =
+      F::soft_margin_loss(input, target, SoftMarginLossOptions());
```

Comment:
We can remove `SoftMarginLossOptions()` in this line to test the "no options" use case :D